src/ewasm/mod.rs

```diff
@@ -0,0 +1,288 @@
+
+                let ewasm_contract = LLVMContract {
+                    contract_declaration,
+                    contract_behaviour_declarations,
+                    struct_declarations,
+                    asset_declarations,
+                    external_traits: external_traits.clone(), // Only cloning the references but is there a better way?
+                    environment: &context.environment,
+                };
+
+                Some(ewasm_contract)
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<LLVMContract>>();
+
+    assert!(!ewasm_contracts.is_empty());
+
+    // It seems from the move target that we are allowed to create multiple contracts, and each of these
+    // gets put into a separate file, so we will do the same here
+    let output_path = Path::new("output");
+    if !output_path.exists() {
+        fs::create_dir(output_path).expect("Could not create output directory");
+    }
+
+    let flint_config =
+        fs::read_to_string(Path::new("flint_config.json")).expect("Could not find flint config");
+    let flint_config = json::parse(flint_config.as_str()).expect("Could not parse flint config");
+    assert!(flint_config.is_object());
+    let llc_path = flint_config
+        .entries()
+        .find(|(key, _)| *key == "llcPath")
+        .expect("No llcPath in flint config")
+        .1
+        .as_str()
+        .expect("Could not find llcPath in flint config");
+    assert!(!llc_path.is_empty());
+
+    let wasm_ld_path = flint_config
+        .entries()
+        .find(|(key, _)| *key == "wasm-ldPath")
+        .expect("No wasm-ldPath in flint config")
+        .1
+        .as_str()
+        .expect("Could not find wasm-ld path in flint config");
+    assert!(!wasm_ld_path.is_empty());
+
+    for contract in ewasm_contracts.iter() {
+        let tmp_path = Path::new("tmp");
+        if !tmp_path.exists() {
+            fs::create_dir(tmp_path).expect("Could not create tmp directory");
+        }
+
+        let file_name = contract.contract_declaration.identifier.token.as_str();
+        let get_path = |folder: &str, ext: &str| format!("{}/{}.{}", folder, file_name, ext);
+
+        // Create LLVM file
+        create_and_write_to_file(
+            Path::new(get_path("tmp", "ll").as_str()),
+            &*generate_llvm(contract).as_bytes(),
+        )
+        .expect("Could not create file");
+
+        // Convert LLVM to wasm32:
+        Command::new(llc_path)
+            .arg("-O3")
+            .arg("-march=wasm32")
+            .arg("-filetype=obj")
+            .arg(get_path("tmp", "ll"))
+            .status()
+            .expect("Could not compile to WASM");
+
+        // Link externally defined functions
+        Command::new(wasm_ld_path)
+            .arg("--no-entry")
+            .arg("--export-all")
+            .arg("--allow-undefined")
+            .arg("-o")
+            .arg(get_path("tmp", "wasm"))
+            .arg(get_path("tmp", "o"))
```

Comment:
Command arguments are not particularly a concern for magic strings, since they are effectively code. The concern is directories. If we use the "tmp" string in two different ways, and then need to rename one of them, which cases do we rename? By using a constants for each we could just rename the constants. Avoid using specific implementation choice strings like this mid-code, instead declare them as constants somewhere relevant. Otherwise you might as well check if `name == "Susan"`

Reply:
To clarify, your problem here is the hard coding of "tmp" as a directory name?